waspc/src/Wasp/Generator/NpmInstall.hs

```diff
@@ -0,0 +1,106 @@
+module Wasp.Generator.NpmInstall
+  ( ensureNpmInstall,
+  )
+where
+
+import Control.Concurrent (Chan, newChan, readChan)
+import Control.Concurrent.Async (concurrently)
+import Control.Monad (when)
+import qualified Data.Aeson as Aeson
+import qualified Data.ByteString.Lazy as B
+import StrongPath (Abs, Dir, File', Path', Rel, relfile, (</>))
+import qualified StrongPath as SP
+import System.Directory (doesFileExist, removeFile)
+import System.Exit (ExitCode (..))
+import Wasp.AppSpec (AppSpec)
+import Wasp.Generator.Common (ProjectRootDir)
+import qualified Wasp.Generator.Job as J
+import Wasp.Generator.Job.IO (printPrefixedJobMessage)
+import Wasp.Generator.Monad (GeneratorError (..), GeneratorWarning (..))
+import qualified Wasp.Generator.NpmDependencies as N
+import Wasp.Generator.ServerGenerator as SG
+import qualified Wasp.Generator.ServerGenerator.Setup as ServerSetup
+import Wasp.Generator.WebAppGenerator as WG
+import qualified Wasp.Generator.WebAppGenerator.Setup as WebAppSetup
+
+-- | Run npm install if needed
+ensureNpmInstall :: AppSpec -> Path' Abs (Dir ProjectRootDir) -> IO ([GeneratorWarning], [GeneratorError])
+ensureNpmInstall spec dstDir = do
+  let appSpecFullStackPackageJsonDependencies = N.buildFullStackNpmDependencies spec SG.waspNpmDependencies WG.waspNpmDependencies
```

Comment:
I would rename this to `errorOrFullStackDeps`.

Then below you can do

```hs
case errorOrFullStackDeps of
  Left errorMsg -> ...
  Right fullStackDeps -> ...
```

instead of having 
```hs
case appSpecFullStackPackageJsonDependencies of
  Left ...
  Right fullStackDependencies ->
```

where you have no good reason for naming it first `appSpecFullStackPackageJsonDependencies` and then `fullStackDependencies`, since it sounds like the same thing but just shortened, but it actually don't the same thing as the first one is possibly an error.